### PR TITLE
[update] Define Clusters & Similarity Calculation with Tests

### PR DESCRIPTION
**Pull request by kingchou007**
```
crowndata-evaluation/
├── crowndata_evaluation/
│   ├── services/
│   │   ├── action_consistency/
│   │   │   ├── tests/
│   │   │   │   ├── test_clustering.py                         
│   │   │   │   └── test_state_similarity.py
│   │   ├── clustering.py
│   │   ├── state_similarity_calculator.py                
```
This PR introduces several improvements and new features related to state similarity and clustering functionality, along with tests to ensure correctness.

### CODE REVIEW

**Comment by kingchou007** [comment]
```
crowndata-evaluation/
├── crowndata_evaluation/
│   ├── services/
│   │   ├── velocity/                                
│   │   │   ├── tests/
│   │   │   ├──...
│   │   ├── action_consistency/
│   │   │   ├── tests/
│   │   │   │   ├── test_clustering.py                         
│   │   │   │   └── test_state_similarity.py
│   │   │   ├── clustering.py
│   │   │   ├── state_similarity_calculator.py  
│   │   │   ├── action_variance_calculator.py

```

@wxj77 Can we organize the main computational methods into different sections? For example, both state similarity and variance are calculated based on clusters, so I’ve grouped them together. Velocity calculations are organized separately. Each folder should correspond to a specific functionality.

File: crowndata-evaluation/crowndata_evaluation/services/action_consistency/state_similarity_calculator.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from typing import Dict, Union
+from clustering import define_clusters
+
+
+class StateSimilarityCalculator:
```
Comment (wxj77) [inline crowndata-evaluation/crowndata_evaluation/services/action_consistency/state_similarity_calculator.py:6]: do we need a class method? def is also fine?

**Comment by wxj77** [comment]
I think it's good to organize like this. But maybe put test in another place. I added an example in the recent develop branch.

In a Poetry project, the recommended structure is to place your test files in a dedicated tests/ directory at the root of your project. This keeps your test code separate from your application code, which helps maintain clarity and organization.
```
my_project/
│
├── my_project/                # Your main project code
│   ├── __init__.py
│   ├── module1.py
│   ├── module2.py
│
├── tests/                     # Your test directory
│   ├── __init__.py
│   ├── test_module1.py         # Tests for module1.py
│   ├── test_module2.py         # Tests for module2.py
│
├── pyproject.toml              # Poetry configuration
├── README.md
├── LICENSE
└── .gitignore
```
```
> crowndata-evaluation/
> ├── crowndata_evaluation/
> │   ├── services/
> │   │   ├── velocity/                                
> │   │   │   ├── tests/
> │   │   │   ├──...
> │   │   ├── action_consistency/
> │   │   │   ├── tests/
> │   │   │   │   ├── test_clustering.py                         
> │   │   │   │   └── test_state_similarity.py
> │   │   │   ├── clustering.py
> │   │   │   ├── state_similarity_calculator.py  
> │   │   │   ├── action_variance_calculator.py
> ```
> 
> @wxj77 Can we organize the main computational methods into different sections? For example, both state similarity and variance are calculated based on clusters, so I’ve grouped them together. Velocity calculations are organized separately. Each folder should correspond to a specific functionality.



> ```
> crowndata-evaluation/
> ├── crowndata_evaluation/
> │   ├── services/
> │   │   ├── action_consistency/
> │   │   │   ├── tests/
> │   │   │   │   ├── test_clustering.py                         
> │   │   │   │   └── test_state_similarity.py
> │   │   ├── clustering.py
> │   │   ├── state_similarity_calculator.py                
> ```
> 
> This PR introduces several improvements and new features related to state similarity and clustering functionality, along with tests to ensure correctness.

**Review by ZhanchongDeng — APPROVED** [review]
LGTM!

**Pull request merged**